app/views/login/__init__.py

```diff
@@ -37,15 +37,17 @@ def load_user_from_request(request):
     if not token:
         return None
 
```

Comment:
And add the login-logic afterwards:

```suggestion

    if token.logins_left > 0:
        token.logins_left -= 1

```